app-typescript/components/Label.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
```

Comment:
Can you use the following interface?

```
interface IProps {
    text: string;
    icon?: 'icon1' | 'icon2'; // <-- add available icons

    // I suggest not exposing a prop for text transform and handling it inside the component
    // small and large would be uppercase and medium - sentence case
    size?: 'small' | 'normal' | 'large';
    color: 'color1' | 'color2'; // <-- add available colors
    style?: 'filled' | 'hollow'; // defaults to filled
    onClick?(): void; // if onClick is provided, a "link" icon is displayed
}
```

Does it make sense to have a icon-only label? If it does, add a separate component like with icon-only buttons. I wasn't able to understand what interface property `types` is referring to. Can you explain?

Reply:
it's impossible to define all colors and icons  because there is a lot of them and there are being constantly updated

Reply:
make it a `string` then, and add a comment with a link to a file where all defined values are listed.

Reply:
here is a link to icons https://ui-framework.superdesk.org/#/components/icons, and to colors https://ui-framework.superdesk.org/#/components/colors

Reply:
Add the links to the interface like so

```
interface IProps {
    // https://ui-framework.superdesk.org/#/components/icons
    icon?: string;
    ...
}

I'm not sure the colors in that link are implemented. If they are, specify the exact format for a color string. For example, is it "PURPLE500" or "purple_500"?